content/blog/detect-pii/index.md

```diff
@@ -0,0 +1,219 @@
+---
+path: '/detect-pii'
+title: "I shouldn't be seeing this: anonymize sensitive data while debugging using NLP"
+date: 2022-10-10T06:00:00.000+00:00
+featured_image: hero.png
+categories: ['Pixie Team Blogs', 'PII', 'Security']
+authors: ['Benjamin Kilimnik']
+emails: ['benjaminkilimnik@gmail.com']
+---
+
+### Introducing a [PII detector for protocol trace data.](https://detect.streamlitapp.com/) Give it a try and [tell us what you think.](https://github.com/pixie-io/pixie/issues/623) It uses a [custom NLP model](https://huggingface.co/beki/en_spacy_pii_distilbert) trained on a [new PII dataset](https://huggingface.co/datasets/beki/privy) for protocol traces.
+
+![](sample_pii_json.png)
+
+It's 10 pm and you're on-call. A few minutes ago, you received a slack message about performance issues affecting users of your application. You sigh, pour yourself some instant coffee, and start pulling up the logs of your [Kubernetes cluster](notion://www.notion.so/Kubernetes.io). By chance, you peek at the latest HTTP request coming through - it's a purchase for foot cream. Not only that, but it has the customer's name, email, and IP address written all over it.
+
+**"Ah," you think to yourself. "I probably shouldn't be seeing this."**
+
+How often in your career have you muttered these words? With so much personal data flowing through applications, it can be all too easy to chance upon sensitive information while debugging issues. Some observability tools, like [Pixie](https://px.dev/), enable users to [redact data sources they know to be sensitive](https://docs.px.dev/reference/admin/deploy-options/#setting-the-data-access-mode). Unfortunately, this solution drops entire categories of data, removing information that may be useful for debugging. To prevent privacy leaks while retaining useful information, developers need a system that finds and redacts only the sensitive parts of each data sample.
```

Comment:
Can probably remove the Pixie link since this is being published on the Pixie blog.

Reply:
Done